### ISSUE TITLE
I am on linux server, shutdown error

### ISSUE THREAD

**Issue by DreamMiu** [ThizThizzyDizzy/tree-feller#76]
core:purpur-1.19.4-1985
system：debian
log:
`[01:29:20] [Server thread/ERROR]: Nag author(s): '[ThizThizzyDizzy]' of 'TreeFeller v1.22.1' about the following: This plugin is not properly shutting down its async tasks when it is being shut down. This task may throw errors during the final shutdown logs and might not complete before process dies.
[01:29:20] [Server thread/ERROR]: Craft Scheduler Thread - 15 - TreeFeller Stacktrace
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/sun.nio.ch.SocketDispatcher.read0(Native Method)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:47)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/sun.nio.ch.NioSocketImpl.tryRead(NioSocketImpl.java:266)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/sun.nio.ch.NioSocketImpl.implRead(NioSocketImpl.java:317)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/sun.nio.ch.NioSocketImpl.read(NioSocketImpl.java:355)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/sun.nio.ch.NioSocketImpl$1.read(NioSocketImpl.java:808)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/java.net.Socket$SocketInputStream.read(Socket.java:966)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/sun.security.ssl.SSLSocketInputRecord.read(SSLSocketInputRecord.java:484)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/sun.security.ssl.SSLSocketInputRecord.readHeader(SSLSocketInputRecord.java:478)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/sun.security.ssl.SSLSocketInputRecord.bytesInCompletePacket(SSLSocketInputRecord.java:70)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/sun.security.ssl.SSLSocketImpl.readApplicationRecord(SSLSocketImpl.java:1465)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:1069)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/java.io.BufferedInputStream.fill(BufferedInputStream.java:244)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/java.io.BufferedInputStream.read1(BufferedInputStream.java:284)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/java.io.BufferedInputStream.read(BufferedInputStream.java:343)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/sun.net.www.http.HttpClient.parseHTTPHeader(HttpClient.java:826)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/sun.net.www.http.HttpClient.parseHTTP(HttpClient.java:761)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1688)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1589)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:529)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:308)
[01:29:20] [Server thread/ERROR]:   TreeFeller-1.22.1.jar//com.thizthizzydizzy.treefeller.TreeFeller.downloadFile(TreeFeller.java:1712)
[01:29:20] [Server thread/ERROR]:   TreeFeller-1.22.1.jar//com.thizthizzydizzy.treefeller.TreeFeller.refreshPatronsList(TreeFeller.java:1684)
[01:29:20] [Server thread/ERROR]:   TreeFeller-1.22.1.jar//com.thizthizzydizzy.treefeller.TreeFeller.access$400(TreeFeller.java:57)
[01:29:20] [Server thread/ERROR]:   TreeFeller-1.22.1.jar//com.thizthizzydizzy.treefeller.TreeFeller$3.run(TreeFeller.java:653)
[01:29:20] [Server thread/ERROR]:   org.bukkit.craftbukkit.v1_19_R3.scheduler.CraftTask.run(CraftTask.java:101)
[01:29:20] [Server thread/ERROR]:   org.bukkit.craftbukkit.v1_19_R3.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:57)
[01:29:20] [Server thread/ERROR]:   com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[01:29:20] [Server thread/ERROR]:   java.base@17.0.8/java.lang.Thread.run(Thread.java:833)`

**Comment by ThizThizzyDizzy** [comment]
Looks like it should be harmless

**Comment by DreamMiu** [comment]
> Looks like it should be harmless

yes, so far nothing has been found to have an impact except when the server is shut down.

**Comment by ThizThizzyDizzy** [comment]
Unable to reproduce on purpur 1.19.4 or 1.21.11.
Also harmless, and the thing that caused it here will be removed in the future
